### PR TITLE
Identity feature

### LINKED ISSUE #351: UI to configure name and email settings for local user?
When committing to Git, it is important to have proper attribution. Git has two settings that must be changed before making an initial commit:

- `name`
- `email`

Please consider adding a UI that helps a user configure their `name` and `email` settings for Git, with particular consideration given to how JupyterLab may be used in multi-user environments.

## Quick references

- https://help.github.com/en/articles/setting-your-username-in-git
- https://help.github.com/en/articles/setting-your-commit-email-address-in-git

### PR DESCRIPTION
**Pull request by fcollonval**
Fixes #351 and will imply simplification in #345.

## User changes

If the user has no `user.name` and no `user.email` defined in Git global configuration, the GitPanel will be replaced by a warning and a button prompting the user for a name and an email.

image

If the user provides them, they will be added as global configuration options in git. Then the extension will behave as previously.

Note;

* The email is not verified to be a valid email string
* If the user removes his identity after opening JupyterLab (very unlikely), the extension is never checking again if an identity is set.

## Backward incompatibilities

None

### CODE REVIEW

**Comment by jaipreet-s** [comment]
Thanks for this. Based on a quick look,  this will disable all of the Git panel functionality and replace it with this error message if theres no identity set. Now, the username/email is not required till the user starts to commit the information, so disabling the "changed/staged" view, and history view, and the upcoming diff functionality seems heavy for not having a username/email set.

My suggestion would be would be to delay this check till the user goes to commit the the information

**Comment by fcollonval** [comment]
Thanks for the feedback @jaipreet-s.

Indeed the current approach is disabling the main panel. I'll update it following your suggestion to check once and only once at the first commit attempt. 
Does that sounds better?

**Comment by jaipreet-s** [comment]
That sounds great! I'll also engage @weihwang  to have a look at the username/email popup wording/UX.

File: jupyterlab_git/git.py [hunk 1]
```diff
@@ -16,6 +16,45 @@ def __init__(self, root_dir, *args, **kwargs):
...
+                response["code"] = p.returncode
+                if p.returncode != 0:
+                    response["command"] = " ".join(cmd)
+                    response["message"] = err.decode("utf-8").strip()
+                    return response
+
+            response["message"] = "\n".join(output).strip()
+        else:
+            cmd = ["git", "config", "--global", "--list"]
+            p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (jaipreet-s) [inline jupyterlab_git/git.py]: We don't need to check for the global configuration, right. As long as user name/email is configured at the repo or global leve, we're good. `git config user.email/name` will return the repo level user name/email or the global one if no override is set.

File: jupyterlab_git/git.py [hunk 2]
```diff
@@ -16,6 +16,45 @@ def __init__(self, root_dir, *args, **kwargs):
...
+        """Get or set Git options.
+        
+        If no kwargs, all options are returned. Otherwise kwargs are set.
+        """
+        response = {"code": 1}
+
+        if len(kwargs):
+            output = []
+            for k, v in kwargs.items():
+                cmd = ["git", "config", "--global", "--add", k, v]
```
Comment (jaipreet-s) [inline jupyterlab_git/git.py]: In my opinion, setting these are the global level and potentially overwriting global configuration is overkill. As long as we set it at the repo level, it should suddice.

**Comment by fcollonval** [comment]
@jaipreet-s does it look better?

File: jupyterlab_git/git.py (hunk 2 above)
Comment (fcollonval) [reply jupyterlab_git/git.py]: I set options at repository level

File: jupyterlab_git/git.py (hunk 1 above)
Comment (fcollonval) [reply jupyterlab_git/git.py]: You are right, I'll do that.

**Comment by mathematicalmichael** [comment]
This is a great idea! 
Am I right in interpreting that it first checks for local user info, then global, if missing? 
Then sets local if so.

Do these changes to .git/config migrate upon push?

File: src/widgets/gitPushPull.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Spinner } from '@jupyterlab/apputils';
 import { Widget } from '@phosphor/widgets';
-import { Git, IGitPushPullResult } from './git';
+import { Git, IGitPushPullResult } from '../git';
```
Comment (jaipreet-s) [inline src/widgets/gitPushPull.ts]: I like this new directory structure. Thanks for moving.

**Comment by jaipreet-s** [comment]
@fcollonval This looks great. 

Looks like there is one merge conflict and an unrelated Travis CI failure though

**Comment by fcollonval** [comment]
@jaipreet-s I finally overcome the CI error. Could you please have a look?

**Pull request merged**